src/index.ts

```diff
@@ -44,27 +53,43 @@ const printHelp = (exitCode: number) => {
   return process.exit(exitCode);
 };
 
-const parseArgs = (args: any[]): ParsedArgs =>
-  args.reduce((acc, arg) => {
+const parseArgs = (
+  args: string[]
+): { parsedArgs: Config; parsedInternalFlags: InternalFlags } => {
+  const parsedArgs = args.reduce((acc, arg) => {
     switch (arg) {
-      case "-h":
-      case "--help":
-        acc.options.help = true;
-        break;
-      case "-V":
-      case "--version":
-        acc.options.version = true;
-        break;
       case "-dr":
       case "--detect-roots":
         acc.options.detectRoots = true;
         break;
       default:
+        if (arg.startsWith("--") || arg.startsWith("-")) break;
         acc.paths.push(arg);
     }
 
     return acc;
-  }, defaults);
+  }, v8.deserialize(v8.serialize(defaultConfig)));
+
+  const parsedInternalFlags = args.reduce((acc, arg) => {
+    switch (arg) {
+      case "-h":
+      case "--help":
+        acc.help = true;
+        break;
+      case "-V":
+      case "--version":
+        acc.version = true;
+        break;
+    }
+
+    return acc;
```

Comment:
Why use `v8` like this here?

Reply:
I needed to deep clone the `defaultConfig`. The reduce assigned nested objects in `defaultConfig`. For example, `defaultConfig.options.detecRoots` were assigned to `true` once `parseArgs` were called. This was an issue because I needed `defaultConfig` to always stay at its default state so I can check if an arg as been passed or not, and take the config file's values depending on that.

Alternatively, I could do `JSON.parse(JSON.stringify(defaultConfig))`.